dpath/__init__.py

```diff
@@ -45,7 +49,22 @@ def _split_path(path: Path, separator: Optional[str] = "/") -> Union[List[PathSe
     else:
         split_segments = path.lstrip(separator).split(separator)
 
-    return split_segments
+    final = []
+    for segment in split_segments:
+        if (options.DPATH_ACCEPT_RE_REGEXP and isinstance(segment, str)
```

Comment:
Flatten line and remove parentheses (line is short enough to comply with PEP8)

Reply:
Hi, 
1) Thanks for your thorough review,
2) I will answer questions first, and push to Github later once most changes are implemented.

This change:OK